### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by betatim**
Add recently merged changes to the change log. I also added links to the PR and a mention of the author. If it doesn't add too much noise I think it is nice to give a shout out to the contributor.

One thing I noticed while looking at [the PRs merged since #426](https://github.com/jupyter/repo2docker/pulls?utf8=%E2%9C%93&q=is%3Apr+sort%3Aupdated-desc+) is that things like work on the docs isn't captured in the changelog. What would be a good way to add this? In my mind the change log is a good way for people to see what has changed (not just code but also new/better docs) and give credit to those doing the work.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -6,14 +6,23 @@ Release date: unknown
...
-
-- Change log added [#426]
+- Build from sub-directory: build the image based on a sub-directory of a
+  repository [#413](https://github.com/jupyter/repo2docker/pull/413)
+  by [@dsludwig](https://github.com/dsludwig)
+- Editable mode: allows editing a local repository from a live container
+  [#421](https://github.com/jupyter/repo2docker/pull/421)
+  by [@evertrol](https://github.com/evertrol)
+- Change log added [#426](https://github.com/jupyter/repo2docker/pull/426)
+  by [@evertrol](https://github.com/evertrol)
```
Comment (choldgraf) [inline CHANGES.rst]:
```suggestion
- Refactoring the documentation for contributors [#453](https://github.com/jupyter/repo2docker/pull/453) by [@choldgraf]
```

File: CHANGES.rst [hunk 2]
```diff
@@ -6,14 +6,23 @@ Release date: unknown
...
```
Comment (choldgraf) [inline CHANGES.rst]:
```suggestion

[@evertrol]: https://github.com/evertrol
[@choldgraf]: https://github.com/choldgraf
```

File: CHANGES.rst [hunk 3]
```diff
@@ -6,14 +6,23 @@ Release date: unknown
...
 
-- Editable mode: allows editing a local repository from a live container [#421]
-
-- Change log added [#426]
+- Build from sub-directory: build the image based on a sub-directory of a
+  repository [#413](https://github.com/jupyter/repo2docker/pull/413)
+  by [@dsludwig](https://github.com/dsludwig)
+- Editable mode: allows editing a local repository from a live container
+  [#421](https://github.com/jupyter/repo2docker/pull/421)
+  by [@evertrol](https://github.com/evertrol)
```
Comment (choldgraf) [inline CHANGES.rst]:
```suggestion
  by [@evertrol]
```

File: CHANGES.rst [hunk 4]
```diff
@@ -6,14 +6,23 @@ Release date: unknown
...
+- Change log added [#426](https://github.com/jupyter/repo2docker/pull/426)
+  by [@evertrol](https://github.com/evertrol)
```
Comment (choldgraf) [inline CHANGES.rst]:
```suggestion
  by [@evertrol]
```

**Comment by betatim** [comment]
Just realised: the change log is rst not markdown. I'll update the syntax and find the right way to only list user's URLs once.

**Comment by betatim** [comment]
Syntax updated and implemented your changes @choldgraf. Merge away!

**Comment by choldgraf** [comment]
:shipit:

**Pull request merged**